contracts/Canopy.sol

```diff
@@ -161,31 +161,73 @@ contract Canopy {
     
     function rankPosts() internal {}
 
-    // @dev read from mapping
-    function getPostsByTime() external {}
+    // @dev read from mapping - show 100 of the most recent
+    function getMostRecentPost() external {
+        //since IDs are sequential, length of array = most recent post id
+        return posts.length;
+    }
 
     // @dev read from mapping
-    function getScoreById() internal {}
+    function getScoreById(uint _postID) internal {
+        return posts[_postID].score;
+    }
 
     // @dev user can choose when to cash out, make sure to check address
-    function cashOut() external payable onlyPoster {}
+    function cashOut(uint _postId) external payable onlyPoster {
+        //check that payout is to posterAddress with onlyPoster
+        //Setting the Variables required
+        uint _paymentToPoster;
+        uint _bonusPayout;
+        uint _bonusPayoutVoters;
+        uint _totalBonus;
+        uint _totalValue;
+        uint _totalPayout;
+
+        _totalValue = p.valuePositive + p.valueNegative;
+        Post memory p = posts[_postId];
+        _paymentToPoster = p.stake * (p.valuePositive / totalValue) + p.valuePositive;
+        //check that payout is not more than 50% pool balance and stake amount
+        if(poolValue * 0.50 > p.stake) {
+            _bonusPayout = (p.stake * 0.50);
+            _bonusPayoutVoters = (p.stake * 0.50);
+            _totalBonus = _bonusPayout + _bonusPayoutVoters;
+            return _totalBonus;
+        } else if (poolValue * 0.50 <= p.stake) {
+            _totalBonus = 0;
+            return _totalBonus;
+        }  
+        
+        _totalPayout = _paymentToPoster + _totalBonus;
+
+        //check valuePositive is greater than 0.75 of TotalValue
+        require(p.valuePositive >= (0.75*totalValue), "Not a good post");
+        //check that poolValue is positive
+            if(poolValue.balance > 0) {
```

Comment:
if there's no money in the contract then the rest of this is pointless! let's do this before we start any payment logic (so we don't waste gas), and let's make _this_ a require statement.